include/bluetooth/audio/has.h

```diff
@@ -107,6 +121,82 @@ int bt_has_client_discover(struct bt_conn *conn);
  */
 int bt_has_client_conn_get(const struct bt_has *has, struct bt_conn **conn);
 
+/** @brief Register structure for preset. */
+struct bt_has_preset_register_param {
+	/**
+	 * @brief Preset index.
+	 *
+	 * Unique preset identifier. The value shall be other than 0.
+	 */
+	uint8_t index;
+
+	/**
+	 * @brief Preset properties.
+	 *
+	 * Bitfield of preset properties.
+	 */
+	enum bt_has_properties properties;
+
+	/**
+	 * @brief Preset name.
+	 *
+	 * Preset name that further can be changed by either server or client if
+	 * @kconfig{CONFIG_BT_HAS_PRESET_NAME_DYNAMIC} configuration option has been enabled.
+	 */
+	const char *name;
+};
+
+/**
+ * @brief Register preset.
+ *
+ * Register preset. The preset will be a added to the list of exposed preset records.
+ *
+ * @param param Preset registration parameter.
+ *
+ * @return 0 if success, errno on failure.
+ */
+int bt_has_preset_register(struct bt_has_preset_register_param *param);
+
+/**
+ * @brief Unregister Preset.
+ *
+ * Unregister preset. The preset will be removed from the list of preset records.
+ *
+ * @param index The index of preset that's being requested to unregister.
+ *
+ * @return 0 if success, errno on failure.
+ */
+int bt_has_preset_unregister(uint8_t index);
+
+enum {
+	BT_HAS_PRESET_ITER_STOP = 0,
+	BT_HAS_PRESET_ITER_CONTINUE,
+};
+
+/**
+ * @typedef bt_has_preset_func_t
+ * @brief Preset iterator callback.
+ *
+ * @param index The index of preset found.
+ * @param properties Preset properties.
+ * @param name Preset name.
+ * @param user_data Data given.
+ *
+ * @return BT_HAS_PRESET_ITER_CONTINUE if should continue to the next preset.
+ * @return BT_HAS_PRESET_ITER_STOP to stop.
+ */
+typedef uint8_t (*bt_has_preset_func_t)(uint8_t index, enum bt_has_properties properties,
+					const char *name, void *user_data);
+
+/**
+ * @brief Preset iterator.
+ *
+ * @param index Preset index, passing 0 skips index matching.
```

Comment:
As described, passing 0 skips index matching. Thus if index is non-null it will look for preset with given index to call callback. I've added short description to make it more clear

Reply:
I see. 

So basically if `index == 0`, it does a foreach on... Well each preset. 
But if `index != 0` it does a lookup on a single preset?

Reply:
Exactly, that is how it should work :D But there was a bug you catch few comments below ([index](https://github.com/zephyrproject-rtos/zephyr/pull/44679#discussion_r846011814))

Reply:
OK that makes sense :) 

The alternative is of course to have a `index`-less foreach, and `lookup_has_by_index` function :)